### PR TITLE
🛠️ Fix ➾ Implement Typescript Generation for Lambda

### LINKED ISSUE #1292: 🐛 Bug ➾ plugin-contract-types : Module '"./type-aliases"' has no exported member 'lambda'.
### 🆘 What happened?

image

When generating classes from a smartcontract storage containing a lambda field :  

- on the generated file pokeGame.types.ts : lambda is correctly imported but not found
- on the generated file type-aliases.ts : lamba is not declared

during compilation, it fails with image above

### 🪵 Relevant log output

```shell
ERROR in src/pokeGame.types.ts:3:19
TS2305: Module '"./type-aliases"' has no exported member 'lambda'.
    1 |
    2 | import { ContractAbstractionFromContractType, WalletContractAbstractionFromContractType } from './type-utils';
  > 3 | import { address, lambda, MMap, nat, ticket } from './type-aliases';
      |                   ^^^^^^
    4 |
    5 | export type Storage = {
    6 |     feedback: string;
```

### 🐘 How impactful is this bug?

🚒 Showstopper

### ⏱️ Prevalance

_No response_

### 💻 Operating System

_No response_

### 🕸️ System Architecture

_No response_

### 🌮 Taqueria Version

v0.19.3

### 🌿 Node.js Version

_No response_

### ☎️ Contact Details

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by AlirezaHaghshenas**
# 🌮 Taqueria PR

Fixes #1292 

## 🪁 Description

This PR adds support for the Lambda type in the contract types plugin

## 🪂 Pre-Merge Checklist (Definition of Done)

🚦 Required to merge:

- [ ] ⛵ My code builds cleanly, and I have manually tested the changes
- [ ] 🏄‍♂️ Another team member has built this branch and done manual testing on the change
- [ ] 🏖️ New and existing unit tests pass locally and in CI
- [ ] 🔱 The test plan has been implemented and verified by an SDET
- [ ] ⛱️ I have completed this PR template in full and updated the title appropriately
- [ ] 🦀 Automated tests have been written and added to this PR
- [ ] 🐬 I have commented my code, particularly in hard-to-understand areas

## 🎢 Test Plan

_Please describe the testing strategy and plan for this PR. Keep this lightweight and anticipate any testing challenges_

## 🛸 Type of Change

- [ ] 🧽 Chore ➾
- [x] 🛠️ Fix ➾
- [ ] ✨ Feature ➾
- [ ] 👷 Refactor ➾
- [ ] 🧪 Pre-Release ➾
- [ ] 🚀 Release ➾

### CODE REVIEW

**Review by ricklove — COMMENTED** [review]
Looks right, just need to add e2e tests and add some contracts and expected output:

https://github.com/ecadlabs/taqueria/blob/1e667b348309fa1b656a66f4dc94a91376384fee/tests/e2e/taqueria-plugin-contract-types.spec.ts#L235

**Comment by jchenche** [comment]
Don't we need to define `lambda` in `taqueria-plugin-contract-types/src/type-aliases.ts`?

**Comment by AlirezaHaghshenas** [comment]
> Don't we need to define `lambda` in `taqueria-plugin-contract-types/src/type-aliases.ts` to be exported?

No, we don't use lambda directly in ts. We use arrow signature: `(args: argType) => retType` which does not need any type alias.

Unless I'm not aware of something

**Comment by ricklove** [comment]
> Don't we need to define `lambda` in `taqueria-plugin-contract-types/src/type-aliases.ts` to be exported?

Ah right: this will probably need additional generics added to Taquito, but we will need something here:

type contractTypes = { methods: Methods, methodsObject: MethodsObject, storage: Storage, code: { __type: 'ExampleLambdaCode', protocol: string, code: object[] } };

**Comment by ricklove** [comment]
Hmm, actually the taquito generics might work, we might just need to modify this file to put the lambda type in the correct place (views?)

**Comment by ricklove** [comment]
We also need to add a usage example here which will help verify if it is working right:

https://github.com/ecadlabs/taqueria/blob/refactor-protocal/taqueria-plugin-contract-types/example/example-usage.ts#L6

**Comment by hu3man** [comment]
Please update the title of this PR with the icon and proper capitalization when you have time @AlirezaHaghshenas

**Comment by AlirezaHaghshenas** [comment]
@ricklove 

> Hmm, actually the taquito generics might work, we might just need to modify this file to put the lambda type in the correct place (views?)

I actually don't know what you're talking about. I might need more context

**Review by hu3man — APPROVED** [review]
Great work @AlirezaHaghshenas!

File: taqueria-plugin-contract-types/src/generator/typescript-output.ts [hunk 1]
```diff
@@ -223,6 +232,9 @@ ${tabs(indent)}}])`;
 		if (t.kind === `unknown`) {
 			return `unknown`;
 		}
+		if (t.kind === 'lambda') {
+			throw new GenerateApiError(`Don't know how to generate default value for lambda`, { t });
```
Comment (ricklove) [inline taqueria-plugin-contract-types/src/generator/typescript-output.ts:236]: Hmm, this warrants a new helper in the tas utils, so this should be something like this:

```
return `tas.intruction('')`
```

Also, this needs to be added to the tas code to convert a string to a MichelsonInstruction

**Review by ricklove — CHANGES_REQUESTED** [review]
Need to add a helper to create a lambda

**Review by ricklove — APPROVED** [review]
Nice!

**Pull request merged**